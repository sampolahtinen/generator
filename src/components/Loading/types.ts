export interface LoadingProps {
  children: any;
}
