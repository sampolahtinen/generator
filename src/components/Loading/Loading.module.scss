@import '../../styles/colors.scss';

.loading {
  position: relative;
  width: 100%;
  text-align: center;
  margin: 10rem;
  span {
    color: $red;
    margin: auto;
    opacity: 0.3;
    -webkit-animation: loader 800ms infinite ease-out 0.5s;
    -moz-animation: loader 800ms infinite ease-out 0.5s;
    -o-animation: loader 800ms infinite ease-out 0.5s;
    -ms-animation: loader 800ms infinite ease-out 0.5s;
    animation: loader 800ms infinite ease-out 0.5s;
  }
}

@-webkit-keyframes loader {
  0% {
    color: $red;
    opacity: 0.3;
  }
  50% {
    color: $red;
    opacity: 0.5;
  }
  100% {
    color: $red;
    opacity: 1;
  }
}

@-moz-keyframes loader {
  0% {
    color: $red;
    opacity: 0.3;
  }
  50% {
    color: $red;
    opacity: 0.5;
  }
  100% {
    color: $red;
    opacity: 1;
  }
}

@-ms-keyframes loader {
  0% {
    color: $red;
    opacity: 0.3;
  }
  50% {
    color: $red;
    opacity: 0.5;
  }
  100% {
    color: $red;
    opacity: 1;
  }
}

@-o-keyframes loader {
  0% {
    color: $red;
    opacity: 0.3;
  }
  50% {
    color: $red;
    opacity: 0.5;
  }
  100% {
    color: $red;
    opacity: 1;
  }
}

@keyframes loader {
  0% {
    color: $red;
    opacity: 0.3;
  }
  50% {
    color: $red;
    opacity: 0.5;
  }
  100% {
    color: $red;
    opacity: 1;
  }
}
