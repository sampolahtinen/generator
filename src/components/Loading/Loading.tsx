import React from 'react';
import { LoadingProps } from './types';
import styles from './Loading.module.scss';

const Loading = ({ children }: LoadingProps) => (
  <div className={styles.loading}>
    <span>{children}</span>
  </div>
);

export default Loading;
