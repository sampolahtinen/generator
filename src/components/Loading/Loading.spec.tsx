import React from 'react';
import { shallow } from 'enzyme';
import Loading from './Loading';

describe('Loading Component', () => {
  const wrapper = shallow(<Loading>Loading</Loading>);

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders text passed as children', () => {
    expect(wrapper.find('.loading').text()).toEqual('Loading');
  });
});
