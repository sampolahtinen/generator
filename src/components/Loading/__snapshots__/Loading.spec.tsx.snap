// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component renders 1`] = `
<div
  className="loading"
>
  <span>
    Loading
  </span>
</div>
`;
